src/diagnostics.rs

```diff
@@ -130,23 +133,23 @@ impl Diagnostic for DocDiagnostic {
   }
   fn snippet_fixed(&self) -> Option<DiagnosticSnippet<'_>> {
     match &self.kind {
-      DocDiagnosticKind::PrivateTypeRef {
-        reference_location, ..
-      } => Some(DiagnosticSnippet {
-        source: Cow::Borrowed(&self.text_info),
-        highlight: DiagnosticSnippetHighlight {
-          style: DiagnosticSnippetHighlightStyle::Hint,
-          range: DiagnosticSourceRange {
-            start: DiagnosticSourcePos::ByteIndex(
-              reference_location.byte_index,
-            ),
-            end: DiagnosticSourcePos::ByteIndex(
-              reference_location.byte_index + 1,
-            ),
+      DocDiagnosticKind::PrivateTypeRef(diagnostic) => {
+        Some(DiagnosticSnippet {
+          source: Cow::Borrowed(&diagnostic.reference_text_info),
```

Comment:
Fix is here. It should be using the referenced module's text info and not the original module's text info